src/Apps/Sell/Utils/previousSubmissionUtils.ts

```diff
@@ -18,20 +20,34 @@ export const usePreviousSubmission = () => {
   const [submissionID, setSubmissionID] = useState<string | null>(null)
   const [step, setStep] = useState<SellFlowStep | null>(null)
 
+  const stepSchema = Yup.string().oneOf([...STEPS])
+  const idSchema = Yup.string().required()
+
+  const isStepValid = (step: any): step is SellFlowStep | null =>
+    stepSchema.isValidSync(step)
+
+  const isIdValid = (id: any): id is string => idSchema.isValidSync(id)
```

Comment:
You can move these out of the function body. Which should also avoid your dependency array ignore.

Reply:
And you can use validateSync instead of isValidSync which should just return the correct type

Reply:
(Rather than using it as a typeguard)